packages/taboule/src/state/queries.ts

```diff
@@ -0,0 +1,60 @@
+import { APIError } from "@shared/errors/APIError";
+import { ChannelRelated } from "@shared/models/ChannelRelated";
+import { GetAPI } from "@shared/providers/api.provider";
+import { available, queryStrict } from "avenger";
+import { CachedQuery } from "avenger/lib/Query";
+import { SearchQuery } from "@shared/models/http/SearchQuery";
+import { pipe } from "fp-ts/lib/function";
+import * as TE from "fp-ts/lib/TaskEither";
+
+export interface SearchRequestInput {
+  Params: any;
+  Query: SearchQuery;
+}
+
+export interface Results<T> {
+  total: number;
+  content: T[];
+}
+
+type EndpointQuery<C> = CachedQuery<SearchRequestInput, APIError, Results<C>>;
+
+export interface TabouleQueries {
+  ccRelatedUsers: EndpointQuery<ChannelRelated>;
```

Comment:
To add a new api callable from the component you need to add the query definition here add theimplementation in `GetDataTableQueries` @vecna